### ISSUE TITLE
Nouvelle facette serving_size="serving" pour data-quality

### ISSUE THREAD

**Issue by TaciteOFF** [openfoodfacts/openfoodfacts-server#5163]
Bonjour,

Il semblerait que quelques utilisateurs (ou applications) indiquent parfois par erreur "serving" comme taille de portion. 
Une facette pour détecter et corriger ça pourrait être utile.

Merci

**Comment by teolemon** [comment]
See also #1008

**Comment by stephanegigandet** [comment]
corresponding code is in DataQualityFood.pm

**Comment by CharlesNepote** [comment]
* There are [more than 11,200+ products](https://mirabelle.openfoodfacts.org/products/simplified?_facet=serving_size&serving_size=serving&_facet=creator) with "serving" as `serving_size`. It is not brought by a particular app, it's just users.
* [1900 of 11,200 do not have fat value](https://mirabelle.openfoodfacts.org/products/all?_sort=rowid&_facet=serving_size&_facet=creator&_facet=data_quality_errors_tags&fat_100g__isnull=1&serving_size__exact=serving) (and probably no nutritional values at all), so `serving_size` containing "serving" should be deleted in this case
* Probably many products with "serving" have nutrition facts for 100g and not for a serving size.
* [990+ of them have a data quality error already detected](https://mirabelle.openfoodfacts.org/products?sql=--+count+number+of+products+with+a+data+quality+errors+which+serving_size+is+%22serving%22%0D%0Aselect+--+rowid%2C+code%2C+url%2C+creator%2C+created_datetime%2C+last_modified_datetime%2C+product_name%2C+serving_size%0D%0Acount%28data_quality_errors_tags%29+as+data_quality_errors_related_to_serving_size_equal_serving%0D%0Afrom+%5Ball%5D%0D%0Awhere%0D%0Aserving_size+%3D%3D+%22serving%22%0D%0Aand+data_quality_errors_tags+%21%3D+%22%22%0D%0Alimit+210), `en:nutrition-data-per-serving-serving-quantity-is-0` for nearly half of them

This would lead to increase data quality errors from 5.78% to ~6.16% for an issue that is having no impact on Nutri-Score, Nova, etc. for at least more than 10%.

So I would be in favour to:

- [ ] 1. Delete all the value "serving" in the `serving_size` as:
  - when users are often seeing this value they can think it's a good value
  - it's easier to complete an empty field than a completed one
  - it's not worse to have an empty field than a false one
- [ ] 2. Create the data quality error facet to manage the new ones

Maybe 2 can be done before 1.

**Comment by benbenben2** [comment]
> Create the data quality error facet to manage the new ones
Do you want only the case when serving_size="serving"? 
Or do you want to generalize for all strings? (just thinking there..., we always expect some number in this field - like 20g, 20ml, for example - right? If yes, then, we could raise an error when there is no integer/number in this field. What do you think?

 Just checked with Mirabelle, that could include some values like "-" (6), "une tranche" (1), "une noisette" (1), "une pression par narine" (1), "une cuillere à café" (1), "une biscotte g" (1), "tbsp" (1), "servingg" (1), etc. 
But occurrence of those values (written in parenthesis) is nothing comparable to "serving" (11 199)

**Comment by CharlesNepote** [comment]
@benbenben2 yes why not. There are [227 different values](https://mirabelle.openfoodfacts.org/products?sql=select+code%2C+url%2C+serving_size%2C+%0D%0Acount%28serving_size%29+as+c%0D%0Afrom+%5Ball%5D+%0D%0Awhere+serving_size+REGEXP+%27%5E%5B%5E0-9%5D%2B%24%27%0D%0Agroup+by+serving_size+%0D%0Aorder+by+c+desc+%0D%0Alimit+300) for this field. Most of them are given once but it would be nice to detect it.

The hyphen (`-`) should not be included. I think it could be a good practice to always use it when we want to precise that there is no value (null).

@stephanegigandet can you confirm there is no reason not to have at least one number in this field?

**Comment by stephanegigandet** [comment]
It's a free form field, so we can have anything in it. We could have a warning when there isn't a number, but I wouldn't make it an error.

What would be interesting is a quality warning when we don't have a value for serving size, or we have a value that we can't convert to g/ml, AND we have nutrition facts indicated per serving. In that case, while we have nutrition facts, we can't compute Nutri-Score etc

**Comment by CharlesNepote** [comment]
@stephanegigandet 

> It's a free form field, so we can have anything in it. We could have a warning when there isn't a number, but I wouldn't make it an error.

Do you see some examples where a value without a number would be relevant?
If we look at what the users already entered, in the list I mentioned earlier, there are:
* "Non indiquée" (7), "Non précisé" (6), "Non précisée" (1), "Non spécifié (1)", "non communiqué" (1), "N.A" (1), "Not mentioned" (1), "No indica" (1), "nicht angegeben" (1), etc. => IMHO they should be converted to "-"
* "tbsp" (1) or "cuillère à soupe" (1) (but we're not sure if it's "one" tbsp, so I would say it's an error)
* to me, all the other values appear to be either errors, either values that can't be converted into a size in grams or ml.

So I would tend to conclude that:
* either there are values which should be converted into "-" (manually or automatically)
* either there are values that are not relevant, and, as such, should be "data quality errors"

So I would count all these values as "data quality errors" and, while we don't do it automaticaly, let people manually fix "non communiqué" into "-".

> What would be interesting is a quality warning when we don't have a value for serving size, or we have a value that we can't convert to g/ml, AND we have nutrition facts indicated per serving. In that case, while we have nutrition facts, we can't compute Nutri-Score etc

At least we could do this. But honestly, currently quality warnings are not read by anyone (or so few). And there is currently no easy way to tag a warning as "checked". So for all the reasons I mentioned in this comment, I think it should be a data quality error.

**Comment by stephanegigandet** [comment]
> I think it should be a data quality error.

OK.

**Comment by benbenben2** [comment]
@CharlesNepote, is it solved by this PR https://github.com/openfoodfacts/openfoodfacts-server/pull/8091?
